source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -2603,7 +2603,33 @@ void Graph::AddLink(ed::PinId inputPinId, ed::PinId outputPinId)
                             {
                                 if (uiUpNode->getNode())
                                 {
-                                    pin->_input->setConnectedNode(uiUpNode->getNode());
+                                    mx::NodePtr upstreamNode = _graphNodes[upNode]->getNode();
+                                    mx::NodeDefPtr upstreamNodeDef = upstreamNode->getNodeDef();
+                                    bool isMultiOutput = upstreamNodeDef ? upstreamNodeDef->getOutputs().size() > 1 : false;
```

Comment:
This is extra work since it's not valid to just add the `output` attribute all the time to the downstream input. So we test if the upstream node is multit-output. If so then add the `output` attribute, otherwise, just add the `nodename` reference.